sdk/src/transaction/mod.rs

```diff
@@ -1056,13 +1056,16 @@ impl Transaction {
             return Err(TransactionError::InvalidAccountIndex);
         }
 
-        signers
-            .iter()
-            .enumerate()
-            .for_each(|(i, (pubkey, signature))| {
-                self.signatures[i] = *signature;
-                self.message.account_keys[i] = *pubkey;
-            });
+        for (pubkey, signature) in signers {
+            let index = self
+                .message
+                .account_keys
+                .iter()
```

Comment:
i was optimizing for the other thing, avoiding the `.collect()` :sweat_smile:

Reply:
though now that i think of it, i can do both

Reply:
Because of program ids, `account_keys` is always going to be longer than `signers` in practice. So I would still prefer that `account_keys` be the thing that's only iterated once when finding the indices. I can write a code suggestion, if you want.